### PR TITLE
Compile node modules

### PR DESCRIPTION
**Pull request by kepta**
This enables compilation of `node_modules` by default.
- Add some testing to cover compilation of node_modules
- Implement two different babel configurations, one for user's src code and the other for node_modules
- Disable use of `.babelrc` and instead, use `babel.config.js`.

@davidtheclark for review, please.

### CODE REVIEW

File: __tests__/__snapshots__/no-config.test.js.snap [hunk 1]
```diff
@@ -32,14 +32,14 @@ Object {
               "extension": ".js",
               "name": "polyfill.chunk.js",
               "path": "<TEMP_DIR>/_underreact-site/underreact-assets/js/polyfill.chunk.js",
-              "size": 97256,
+              "size": 97446,
```
Comment (davidtheclark) [inline __tests__/__snapshots__/no-config.test.js.snap]: Do you know why the file size changed?

File: __tests__/babel.test.js [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+const commandBuild = require('./test-utils/command-build');
+const generateFixture = require('./test-utils/generate-fixture');
+const stubNodeModule = require('./test-utils/create-stub-node-module');
+
+jest.setTimeout(15 * 1000);
+
+const tap = prom =>
+  prom.then(res => {
+    return res;
+  });
```
Comment (davidtheclark) [inline __tests__/babel.test.js]: I don't think I understand the purpose of this utility. How is `tap(x).then(y => {..})` different from `x.then(y => {..})`?

File: __tests__/babel.test.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+const getChunkContent = (tree, type = 'main') => {
```
Comment (davidtheclark) [inline __tests__/babel.test.js]: What is a "chunk" in this context?

File: __tests__/babel.test.js [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+      expect(mainFileContent).toMatch(/=> 'UNDERREACT_TEST_RETURN_VALUE'/);
+    });
+});
+
+test('Reads browserslist from package.json', () => {
+  const fixture = generateFixture({
+    src: {
+      'entry.js': `
+            window.foo = () => 'UNDERREACT_TEST_RETURN_VALUE';
+          `
```
Comment (davidtheclark) [inline __tests__/babel.test.js]: What's going on with the tabbing in these template literals?

File: __tests__/babel.test.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    },
+    'babel.config.js': `
+        module.exports = {
+            presets: ['${babelPresetMapbox}'],
+            plugins: ['${babelPluginFlowStrip}']
+        }
+      `
+  }).then(dirPath => commandBuild({ cwd: dirPath }));
+
+  expect(build).resolves.toMatch(/Using an external Babel config/);
```
Comment (davidtheclark) [inline __tests__/babel.test.js]: I think you need to `return` uses of `resolves` and `rejects` in order to properly handle errors.

File: lib/default-underreact.config.js [hunk 6]
```diff
@@ -1,14 +1,27 @@
 'use strict';
+
 const path = require('path');
+const browserslist = require('browserslist');
+
+const defaultBrowserslistQuery = [
+  '>0.2%',
+  'not dead',
+  'not ie <= 11',
```
Comment (davidtheclark) [inline lib/default-underreact.config.js]: Aren't we supporting IE 11 by default?

File: lib/webpack-config/babel-loader-config.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    test: /\.js$/,
+    // We want to avoid sending babel/runtime back to babel.
+    exclude: /@babel(?:\/|\\{1,2})runtime/,
+    loader: require.resolve('babel-loader'),
+    options: {
+      presets: [
+        [
+          require('../../packages/babel-preset-mapbox'),
+          {
+            override
```
Comment (davidtheclark) [inline lib/webpack-config/babel-loader-config.js]: What's going on here? A comment would help.

File: lib/webpack-config/create-webpack-config.test.js [hunk 8]
```diff
@@ -54,27 +55,21 @@ test('Uses babelrc if it exists at project root', () => {
...
-        loader: require.resolve('babel-loader'),
-        options: {
-          cacheDirectory: true,
-          babelrc: true,
-          compact: false
-        }
-      }
-    ]);
+      ).options.configFile
+    ).toEqual(path.join(tempDir, 'babel.config.js'));
```
Comment (davidtheclark) [inline lib/webpack-config/create-webpack-config.test.js]: Primitive comparison should use `toBe`.

File: __tests__/babel.test.js (hunk 2 above)
Comment (kepta) [reply __tests__/babel.test.js]: Oops, my bad! This is a left over from debugging the tests. I will go ahead and remove this.

File: __tests__/__snapshots__/no-config.test.js.snap (hunk 1 above)
Comment (kepta) [reply __tests__/__snapshots__/no-config.test.js.snap]: I can confirm that this is nothing but harmless jumbling, mostly whitespaces, due to `node_modules` compilation.

File: lib/default-underreact.config.js (hunk 6 above)
Comment (kepta) [reply lib/default-underreact.config.js]: I have fixed this.

**Comment by kepta** [comment]
Going to go ahead and merge this PR.

**Pull request merged**